### PR TITLE
Asynchronous list iterator

### PR DESCRIPTION
**Pull request by JakubFrejlach**
This PR introduces the async `retrieve_list_iterator_async` session operation which leverage the `aiohttp` library for async HTTP communication.

This is the first introduction of the async stuff to the bindings and I suppose it will be refined and revisited in the future. First thing which might be the current drawback is the fact, that the async is not exposed to the user and thus the response is asynchronously obtained and only then further work may be done with the data.

### CODE REVIEW

File: component_registry_bindings/session.py [hunk 1]
```diff
@@ -239,6 +247,37 @@ def retrieve_list_iterator(self, *args, **kwargs):
...
+
+    async def __retrieve_list_async(self, *args, **kwargs):
+        if "list" in self.allowed_operations:
+            method_module = self.__get_method_module(
+                resource_name=self.resource_name, method="list"
+            )
+            async_fn = get_async_function(method_module)
+
+            kwargs.pop("offset", None)
+            limit = kwargs.pop("limit", 100)
```
Comment (JimFuller-RedHat) [inline component_registry_bindings/session.py:266]: can we change default to 50 ?

File: component_registry_bindings/iterators.py [hunk 2]
```diff
@@ -95,7 +96,7 @@ def make_response_iterable(response, retrieve_list_fn, *args, **kwargs):
             kwargs.pop("limit", None)
             kwargs.pop("offset", None)
             param = getattr(response, param_name, None)
-            if param is None:
+            if param in (None, UNSET):
```
Comment (JimFuller-RedHat) [inline component_registry_bindings/iterators.py:99]: guessing this is the temporary guard as prev discussed (until openapi definition is fixed in corgi) ?

**Review by JimFuller-RedHat — APPROVED** [review]
LGTM !

File: component_registry_bindings/iterators.py (hunk 2 above)
Comment (JakubFrejlach) [reply component_registry_bindings/iterators.py:102]: Exactly.

**Pull request merged**